### PR TITLE
Update JWE Response 

### PR DESCRIPTION
**Pull request by mohitc1**
1. Added "Additional Authenticated Data" to JWE response.
2. Update the JweResponse APIs to return byte[]

These changes will be consumed by Session Key part of PRTv3.

Related PR https://github.com/AzureAD/ad-accounts-for-android/pull/2157

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java [hunk 1]
```diff
@@ -69,59 +74,76 @@ public static class JweHeader {
 
     JweHeader mJweHeader;
 
-    String mEncryptedKey;
+    private String mEncryptedKey;
```
Comment (rpdome) [inline common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java:80]: Can we list which fields are non-null, and which fields are nullable? (and in what case they're nullable)?

Alternatively, I would suggest branching this class out into V2 and V3. It's probably easier to detect that way if we're getting NOT a value that is expected in V3 response.

File: common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java [hunk 2]
```diff
@@ -138,4 +160,20 @@ public static JweResponse parseJwe(String jwe) throws JSONException {
...
         return response;
     }
+
+    /***
+     * Helper to perform base64 decoding with logging.
+     * @param input Input string
+     * @param flags
+     * @param failureMessage The message to log in case of failure.
+     */
+    public static byte[] base64Decode(@NonNull final String input, int flags, @NonNull final String failureMessage) {
```
Comment (rpdome) [inline common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java]: This one should probably go to common4j's StringUtil.

File: common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java (hunk 1 above)
Comment (mohitc1) [reply common4j/src/main/com/microsoft/identity/common/java/platform/JweResponse.java:80]: Annotated the fields and added comment for nullable. I think the class is meant for JWE parsing and help layers above (PRTv2 and PRTv3) read the values. The calling flow would validate if particular field were required and is correct for them.

**Review by shahzaibj — APPROVED** [review]

File: common4j/src/test/com/microsoft/identity/common/java/platform/JweResponseTests.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import org.json.JSONException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link JweResponse}.
+ */
+@RunWith(JUnit4.class)
```
Comment (rpdome) [inline common4j/src/test/com/microsoft/identity/common/java/platform/JweResponseTests.java:38]: Please add tests where the

1. nonnull/nullable values in the response is missing. Do we crash?
2. the response contains unexpected field. Do we crash?
3. Malformed response.

**Review by rpdome — APPROVED** [review]

File: common4j/src/test/com/microsoft/identity/common/java/platform/JweResponseTests.java (hunk 3 above)
Comment (mohitc1) [reply common4j/src/test/com/microsoft/identity/common/java/platform/JweResponseTests.java:38]: Added

**Pull request merged**